### PR TITLE
Fix shp time

### PR DESCRIPTION
**Pull request by rbuffat**
@sgillies I hope you don't mind. I implemented the fix for https://github.com/Toblerity/Fiona/issues/692 proposed by @snorfalorpagus  and replaced "get_gdal_version_num() // 1000000" as this will fail for gdal versions older than 1.10.0.

### CODE REVIEW

File: tests/test_schema.py [hunk 1]
```diff
@@ -197,3 +197,37 @@ def test_normalize_std(x):
...
+            items = [('field1',  field_type)]
+            with fiona.open(name, 'w',
+                            driver="ESRI Shapefile",
+                            schema={
+                                'geometry': 'LineString',
+                                'properties': items}) as c:
+                assert False
+            
+        except DriverSupportError:
+            assert True
```
Comment (sgillies) [inline tests/test_schema.py]: @rbuffat would you be willing to change the above in two ways? 1) the several field types could be done using pytest.mark.parametrize https://docs.pytest.org/en/latest/parametrize.html, which creates separate tests for each field type, and 2) use `with pytest.raises(DriverSupportError)` to assert that an exception is raised.

File: tests/test_schema.py [hunk 2]
```diff
@@ -197,3 +197,37 @@ def test_normalize_std(x):
...
+        items = [('field1',  'time')]
+        with fiona.open(name, 'w',
+                        driver="GPKG",
+                        schema={
+                            'geometry': 'LineString',
+                            'properties': items}) as c:
+            assert False
+        
+    except DriverSupportError:
+        assert True
```
Comment (sgillies) [inline tests/test_schema.py]: Similarly, I'd like to see the geopackage case become its own proper test separate from the shapefile test.

File: tests/test_schema.py (hunk 1 above)
Comment (rbuffat) [reply tests/test_schema.py]: I refactored the test using pytest.raises and parametrize

File: tests/test_schema.py (hunk 2 above)
Comment (rbuffat) [reply tests/test_schema.py]: Refactored to a seperate test case

**Comment by rbuffat** [comment]
@sgillies  Done. I also refactored test_schema_ordering_items to use pytest.raises.

Strange that the coverage decreases with the additional tests. Especially that the now missing line should not be covered by tests previously as we don't test for old gdal 1.x versions.

**Comment by sgillies** [comment]
@rbuffat I don't have 100% trust in the math at coveralls.io. Their issue tracker has a number of similar reports.

Would you be willing to rebase your changes on the maint-1.8 branch? I wish I had spotted this before. Sorry!

**Comment by rbuffat** [comment]
@sgillies could you give me some pointers on how to do the rebase correctly? "git rebase upstream/maint-1.8" did not really do what I intended

**Pull request closed without merging**